pkg/kubelet/nodeconfig/initconfig.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	yaml "k8s.io/apimachinery/pkg/util/yaml"
+	api "k8s.io/kubernetes/pkg/api"
+	ccv1a1 "k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
+)
+
+// initConfigExists is a simple existential check for the init config.
```

Comment:
Once the KubeletConfiguration type is decomposed into sub-objects, it is possible that this dir will contain a file for each sub-object. initConfigExists indicates that the user intends to provide an init config; validating that they provided an init config *correctly* happens when we load it.